core/commons/che-core-commons-mail/src/test/java/org/eclipse/che/mail/MailSenderTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.mail;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import com.dumbster.smtp.SimpleSmtpServer;
+import com.dumbster.smtp.SmtpMessage;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Map;
+import org.testng.ITestContext;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class MailSenderTest {
+  private MailSender mailSender;
+  private SimpleSmtpServer server;
+
+  public static void assertMail(
+      SimpleSmtpServer server,
+      String from,
+      String to,
+      String replyTo,
+      String subject,
+      String mimeType,
+      String body,
+      String attachmentContentID,
+      String attachmentFileName) {
+    assertEquals(server.getReceivedEmails().size(), 1);
+    SmtpMessage email = server.getReceivedEmails().iterator().next();
+
+    assertEquals(email.getHeaderValue("Subject"), subject);
+    assertEquals(email.getHeaderValue("From"), from);
+    assertEquals(email.getHeaderValue("Reply-To"), replyTo);
+    assertEquals(email.getHeaderValue("To"), to);
+    assertTrue(email.getBody().contains("Content-Type: " + mimeType));
+    assertTrue(email.getBody().contains(body));
+    if (attachmentFileName != null && attachmentContentID != null) {
+      assertTrue(email.getBody().contains("filename=" + attachmentFileName));
+      assertTrue(email.getBody().contains("Content-ID: <" + attachmentContentID + ">"));
+    }
+  }
+
+  @BeforeMethod
+  public void setup(ITestContext context) throws IOException {
+    //      mailSender = new MailSender("/mail-configuration.properties");
+    server = SimpleSmtpServer.start(SimpleSmtpServer.AUTO_SMTP_PORT);
+
+    Map<String, String> mailConfiguration =
+        ImmutableMap.of(
+            "mail.smtp.host",
+            "localhost",
```

Comment:
It looks like an integration test. WDYT?

Reply:
yes, very tiny one.